src/cpr_data_access/parser_models.py

```diff
@@ -247,7 +247,7 @@ def to_string(self) -> str:  # type: ignore
             [text_block.to_string().strip() for text_block in self.text_blocks]
         )
 
-    def detect_and_set_languages(self) -> "ParserOutput":
+    def detect_and_set_languages(self) -> "BaseParserOutput":
```

Comment:
These are the examples I think we may need to change with a typevar – I did this before for documents with `AnyDocument = TypeVar("AnyDocument", bound="BaseDocument")` which silenced errors, but would appreciate validation of whether a similar approach would be the right one here.